src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, CompletableFuture<UserData>> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        // Use computeIfAbsent to initiate the data-fetching CompletableFuture only if absent
+        return cache.computeIfAbsent(userId, id -> fetchUserDataAsync(id));
+    }
+
+    // Asynchronous method to simulate fetching data with delay
+    private CompletableFuture<UserData> fetchUserDataAsync(String userId) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                // Simulate a delay for fetching user data
+                TimeUnit.SECONDS.sleep(1); // Simulate delay without blocking the cache
+            } catch (InterruptedException e) {
+                // Reset interrupt flag and throw RuntimeException to indicate cancellation
+                Thread.currentThread().interrupt();
```

Comment:
It's good practice to restore the interrupted status by calling `Thread.currentThread().interrupt()`, but it should be followed by a mechanism to stop the operation gracefully rather than throwing a RuntimeException, which may be too aggressive. Consider handling the interruption more gracefully if this can affect other parts of the application.